libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -401,6 +458,26 @@ class MapboxNavigation(
         }
     }
 
+    private fun createInternalNavigationSessionObserver() = object : NavigationSessionObserver {
```

Comment:
Do we need this observer? We can keep providing faster route updates even if we are not navigating. This can still be useful to keep updating the UI and the decision to do so can be made by the user.

Reply:
Yes. To me it doesn't make sense to make faster route request unless you start active guidance. We already provide alternative route at that moment to allow the user to make the decision about which route is faster. I think what you would probably want here is to fire route refresh the moment you make a route request. `RouteRefresh` gives you better traffic updates, ETA's etc. which you might wanna reflect on all the routes. If you make faster route request with the same origin there is almost no chance that we get any different route. 

Hence, based on these discussions we agreed to query for faster route only when you have started navigation. This may change later, as I heard that there are talks about emitting `RouteProgress` during free drive.

Reply:
`DirectionsSession` should preferably be separated as much as possible from the `TripSession`. By depending on `NavigationSession` to request a faster route, we are again tying them together. We can let the user make a choice of whether or not to accept a faster route if `TripSession` is stopped.

Reply:
I don't think we are tying `DirectionsSession` and `TripSession` here. We are just invoking some functionality from `directionSession` only if certain changes happen in the `navigationSession`. If not now, there will be n number of use cases that will come up later where features would want to know the current state of navigation. I have experienced that as a SDK user many times. I don't think exposing navigation states is same as exposing internal mechanisms. The user/end client already knows what state they are in - `FREE_DRIVE`, `ACTIVE_GUIDANCE`, `IDLE`. All navigation state is doing is just giving them an opportunity to listen to those changes.

Reply:
Created [ticket](https://github.com/mapbox/navigation-sdks/issues/242) to explore how we may want to modify it in the next iteration.